### PR TITLE
Fixing radioButton checked status

### LINKED ISSUE #13975: Cannot set the checked property of the radio buttons
This is due to the latest changes that I have made in the radio buttons. Because of a faulty check, I cannot update the 'checked' property of radioButtons through code.

example:

radioButton.checked = true 

will not work.

### PR DESCRIPTION
**Pull request by aasimkhan30**
There was an issue where updating the 'checked' property of radio Button was not working.
This PR fixes #13975

### CODE REVIEW

File: src/sql/base/browser/ui/radioButton/radioButton.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export class RadioButton extends Widget {
 	}
 
 	public set checked(val: boolean) {
-		if (this.inputElement.checked !== this._internalCheckedStateTracker) {
+		if (val !== this._internalCheckedStateTracker) {
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/radioButton/radioButton.ts:77]: Hmm, I feel like we should be doing two things here : 

1. Check if `val !== this._internalCheckedStateTracker`. If that's true then update the internal state tracker and fire the event
2. Check if `this.inputElement.checked !== this._internalCheckedStateTracker)` If that's true then set inputElement.checked to val (which due to 1 we know is the same as the internal state tracker)

This way we'll ensure that any time we set checked that everything is synced up correctly - right now with your change if for some reason the internal state is different than the actual checked value (which would happen if something changed the value directly through code).

Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/radioButton/radioButton.ts:77]: So I think we have another bug here too - it doesn't look like this is updating the internal state when the button is clicked if it's not part of a group. We have the click handler but that only fires off the change event if the button has a name - we should always be firing the change event for the button itself regardless.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]
I still think it might not be a terrible idea to do what I suggest about splitting up the checks. But with the change event firing now I think that my immediate concern shouldn't be a problem anymore so it's probably fine.

File: src/sql/base/browser/ui/radioButton/radioButton.ts (hunk 1 above)
Comment (aasimkhan30) [reply src/sql/base/browser/ui/radioButton/radioButton.ts:79]: @Charles-Gagnon please hold off. This is still broken

Comment (aasimkhan30) [reply src/sql/base/browser/ui/radioButton/radioButton.ts:69]: @Charles-Gagnon, I have fixed this scenario. Please review my PR again.

File: src/sql/base/browser/ui/radioButton/radioButton.ts [hunk 2]
```diff
@@ -109,4 +109,11 @@ export class RadioButton extends Widget {
 	public blur(): void {
 		this.inputElement.blur();
 	}
+
+	private propogatedSetCheck(): void {
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/radioButton/radioButton.ts:113]: `propagatedSetCheck`

But also I'm not sure this is a clear enough name. Maybe something like `updateInternalState` instead...

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**